disnake/onboarding.py

```diff
@@ -0,0 +1,239 @@
+        )
+
+    @property
+    def default_channels(self) -> List[GuildChannel]:
+        """List[:class:`abc.GuildChannel`]: The list of channels that will be shown to new members by default."""
+        return list(filter(None, map(self.guild.get_channel, self._default_channel_ids)))
+
+
+class OnboardingPromptOption(Hashable):
+    """Represents an onboarding prompt option.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The prompt option's ID.
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`]]
+        The prompt option's emoji.
+    title: :class:`str`
+        The prompt option's title.
+    description: :class:`str`
+        The prompt option's description.
+    """
+
+    __slots__ = ("id", "title", "description", "emoji", "guild", "_roles_ids", "_channels_ids")
+
+    def __init__(
+        self,
+        *,
+        title: str,
+        description: str,
+        emoji: Optional[Union[str, PartialEmoji, Emoji]],
+        roles: Sequence[Snowflake],
+        channels: Sequence[Snowflake],
+        guild: Guild,
+    ):
+        # NOTE: The ID may sometimes be a UNIX timestamp since
+        # Onboarding changes are saved locally until you send the API request (that's how it works in client)
+        # so the API needs the timestamp to know what ID it needs to create, should we just add a note about it
+        # or "try" to create the ID ourselves?
+        # I'm not sure if this also happens for OnboardingPrompt
+        self.id = 0
+        self.title = title
+        self.description = description
+        self.guild = guild
+        self._roles_ids = [role.id for role in roles]
+        self._channels_ids = [channel.id for channel in channels]
+
+        self.emoji: Optional[Union[PartialEmoji, Emoji]] = None
+        if emoji is None:
+            self.emoji = None
+        elif isinstance(emoji, str):
+            self.emoji = PartialEmoji.from_str(emoji)
+        elif isinstance(emoji, _EmojiTag):
+            self.emoji = emoji
+        else:
+            raise TypeError("emoji must be a str, PartialEmoji, or Emoji instance")
+
+    def __str__(self) -> str:
+        return self.title
+
+    def __repr__(self) -> str:
+        return (
+            f"<OnboardingPromptOption id={self.id!r} title={self.title!r} "
+            f"description={self.description!r} emoji={self.emoji!r} "
+            f"roles={self.roles!r} channels={self.channels!r}>"
+        )
+
+    @classmethod
+    def _from_dict(cls, *, data: OnboardingPromptOptionPayload, guild: Guild) -> Self:
+        name = data["emoji"]["name"]
+        id = data["emoji"]["id"]
+        animated = data["emoji"]["animated"] if "animated" in data["emoji"] else False
+        emoji = PartialEmoji._emoji_from_name_id(
+            name=name,
+            id=int(id) if id else None,
+            animated=animated,
+            state=guild._state,
+        )
```

Comment:
```suggestion
        if emoji_data := data.get("emoji"):
            emoji = guild._state.get_reaction_emoji(emoji_data)
        else:
            emoji = None
```
This is a bit simpler - even if the name doesn't really make sense before #980 is merged, it's fine for now :p